src/azure-cli/azure/cli/command_modules/profile/custom.py

```diff
@@ -143,7 +143,9 @@ def login(cmd, username=None, password=None, tenant=None, scopes=None, allow_no_
     if identity:
         if in_cloud_console():
             return profile.login_in_cloud_shell()
-        return profile.login_with_managed_identity(username, allow_no_subscriptions)
+        return profile.login_with_managed_identity(
+            identity_id=username, client_id=client_id, object_id=object_id, resource_id=resource_id,
+            allow_no_subscriptions=allow_no_subscriptions)
```

Comment:
Do we need to add a param validation that only one of the four (`identity_id`, `client_id`, `object_id` and `resource_id`) has been provided? Or we can add param help msg telling that these four are mutually exclusive

Reply:
Good suggestion. Added the validation logic.